pyuploadcare/resources/file.py

```diff
@@ -353,45 +369,229 @@ def construct_from(cls, file_info):
+
+        It support files smaller than 100MB only. If you want to upload larger files,
+        use Multipart Uploads.
+
+        Args:
+            - file_objects: list of file objects to upload to
+            - store (Optional[bool]): Should the file be automatically stored
+                upon upload. Defaults to None.
+                - False - do not store file
+                - True - store file (can result in error if autostore
+                               is disabled for project)
+                - None - use project settings
+
+        Returns:
+            ``File`` instance
+
+        """
+        files = {
+            os.path.basename(file_object.name or "")
+            or f"file{index}": file_object
+            for index, file_object in enumerate(file_objects)
+        }
 
         response = cls.upload_api.upload(
-            files={"file": file_obj},
-            store=store,
+            files=files,
+            store=cls._format_store(store),
             secure_upload=conf.signed_uploads,
             expire=conf.signed_uploads_ttl,
         )
-        file_ = cls(response["file"])
-        return file_
+        ucare_files = [cls(response[file_name]) for file_name in files]
+        return ucare_files
+
+    @classmethod
+    def multipart_upload(  # noqa: C901
+        cls,
+        file_obj: IO,
+        store: Optional[bool] = None,
+        size: Optional[int] = None,
+        mime_type: Optional[str] = None,
+        callback: Optional[Callable[[UploadProgress], Any]] = None,
+    ) -> "File":
+        """Upload file straight to s3 by chunks.
+
+        Multipart Uploads are useful when you are dealing with files larger than 100MB
+        or explicitly want to use accelerated uploads.
+
+        Args:
+            - file_obj: file object to upload to
+            - store (Optional[bool]): Should the file be automatically stored
+                upon upload. Defaults to None.
+                - False - do not store file
+                - True - store file (can result in error if autostore
+                               is disabled for project)
+                - None - use project settings
+            - size (Optional[int]): file size in bytes.
+                If not set, it is calculated by ``os.fstat``
+            - mime_type (Optional[str]): file mime type.
+                If not set, it is guessed from filename extension.
+            - callback (Optional[Callable[[UploadProgress], Any]]): Optional callback
+                accepting ``UploadProgress`` to track uploading progress.
+
+        Returns:
+            ``File`` instance
+
+        """
+        if size is None:
+            size = cls._get_file_size(file_obj)
+
+        if not mime_type:
+            mime_type, _encoding = mimetypes.guess_type(file_obj.name)
+
+        if not mime_type:
+            raise ValueError(
+                "Unable to determine file mime type, please set manually"
+            )
```

Comment:
why should we fail such uploads?
i believe setting a default mime (`application/octet-stream`) is better than failing

Reply:
done